index.html

```diff
@@ -3026,145 +3004,134 @@ <h3>Add Proof Set/Chain</h3>
+is not a [=map=], an error MUST be raised and SHOULD convey an error type of
+<a href="https://www.w3.org/TR/VC-DATA-MODEL-2.0#PARSING_ERROR">
+PARSING_ERROR</a>.
           </li>
           <li>
-For each known [=cryptographic suite=], run its [=suite selection algorithm=] on |proof|.
-If all of these algorithms return failure, return "invalid". Otherwise, let |suite:VC cryptographic
-suite| be an [=implementation-defined=] choice of one of the non-failure results.
+Let |proof:map| be |securedDocument|.|proof|.
           </li>
           <li>
-If |suite| requires the
-|proof|["`created`"] value, and it does not [=map/exist=],
+If one or more of |proof|.|type|,
+|proof|.|verificationMethod|, and
+|proof|.|proofPurpose| does not [=map/exist=],
 an error MUST be raised and SHOULD convey an error type of
 <a href="#MALFORMED_PROOF_ERROR">MALFORMED_PROOF_ERROR</a>.
-
           </li>
           <li>
 If |expectedProofPurpose| was given, and it does not match
-|proof|["`proofPurpose`"],
+|proof|.|proofPurpose|,
 an error MUST be raised and SHOULD convey an error type of
 <a href="#MISMATCHED_PROOF_PURPOSE_ERROR">MISMATCHED_PROOF_PURPOSE_ERROR</a>.
           </li>
           <li>
-Let |unsecuredDocument| be a copy of |securedDocument| with
-the `proof` value removed.
-          </li>
-          <li>
-Let |transformedData| be the result of running |suite|'s
-[=cryptographic suite/transform=] algorithm on the |unsecuredDocument|.
-        </li>
-          <li>
-Let |hashData| be the result of running |suite|'s
-[=cryptographic suite/hash=] algorithm on the |transformedData|.
-          </li>
-          <li>
-Let |isProofVerified:valid or invalid| be the result of running |suite|'s
-[=cryptographic suite/verify proof=] algorithm on |hashData| and |proof|.
-          </li>
-          <li>
-If |isProofVerified| is "invalid", an error MUST be raised and SHOULD convey an error type of
-<a href="#MALFORMED_PROOF_ERROR">MALFORMED_PROOF_ERROR</a>.
-          </li>
-          <li>
 If |domain| was given, and it does not contain the same [=strings=] as
-|proof|["`domain`"] (treating a single [=string=] as a [=set=] containing just that
-[=string=]), an error MUST be raised and SHOULD convey an error type of <a
+|proof|.|domain| (treating a single [=string=] as a [=set=] containing just
+that [=string=]), an error MUST be raised and SHOULD convey an error type of <a
 href="#INVALID_DOMAIN_ERROR">INVALID_DOMAIN_ERROR</a>.
           </li>
           <li>
 If |challenge| was given, and it does not match
-|proof|["`challenge`"],  an error MUST be raised and SHOULD
+|proof|.|challenge|, an error MUST be raised and SHOULD
 convey an error type of
 <a href="#INVALID_CHALLENGE_ERROR">INVALID_CHALLENGE_ERROR</a>.
           </li>
           <li>
+Let |cryptosuiteVerificationResult| be the result of running the
+|cryptosuite|.[=cryptosuite instance/verifyProof=] algorithm with
+|securedDocument| provided as input.
```

Comment:
`verifyProof` is defined to be able to return an error. This algorithm needs to handle that error.

Reply:
Fixed in 229663465813a71b117db52dad797591d73147e3.